crane/container.go

```diff
@@ -122,7 +122,9 @@ func (c *container) Dependencies() *Dependencies {
 	dependencies := &Dependencies{}
 	for _, link := range c.RunParams.Link() {
 		linkParts = strings.Split(link, ":")
-		dependencies.All = append(dependencies.All, linkParts[0])
+		if !dependencies.includesAsKind(linkParts[0], "link") {
```

Comment:
**TL;DR** Fixed. Thank you for your patience! :smiley:

I had to re-test it manually and realised, that I have made mistake while testing the code for the first time (10 days ago). I have considered and tried your variant first, but probably I have built binary for wrong code and it did not work. Since then I had assumed it was not working due to internal logic. Sadly I did not revise the premise when I got more familiar with the code.

It might help to have automated tests in travis to prevent such issues in future along with CI for building binaries.